src/braket/pulse/pulse_sequence.py

```diff
@@ -165,46 +166,57 @@ def set_scale(
         return self
 
     def delay(
-        self, frames: Union[Frame, List[Frame]], duration: Union[float, FreeParameterExpression]
+        self,
+        qubits_or_frames: Union[List[Frame], QubitSet],
```

Comment:
I think moving towards a `List[Union[Qubit, Frame]` may make sense instead of allowing both frames and an optional qubit parameter. This would avoid a mandatory arg that could be expected to be null

Reply:
This makes the unique check trickier but it is doable